omod/src/main/java/org/openmrs/module/messages/resources/ActorResponseRestResource.java

```diff
@@ -0,0 +1,243 @@
+		"2.3.*", "2.4.*", "2.5.*", "2.6.*" })
+public class ActorResponseRestResource extends DelegatingCrudResource<ActorResponse> {
+    
+    protected static final String REASON = "REST web service";
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        if (rep instanceof DefaultRepresentation) {
+            final DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("id");
+            description.addProperty("patient");
+            description.addProperty("actor");
+            description.addProperty("person");
+            description.addProperty("response");
+            description.addProperty("textQuestion");
+            description.addProperty("textResponse");
+            description.addProperty("question");
+            description.addSelfLink();
+            description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            description.addLink("ref", ".?v=" + RestConstants.REPRESENTATION_REF);
+            description.addSelfLink();
+            return description;
+        } else if (rep instanceof FullRepresentation) {
+            final DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("id");
+            description.addProperty("patient");
+            description.addProperty("person");
+            description.addProperty("response");
+            description.addProperty("textQuestion");
+            description.addProperty("textResponse");
+            description.addProperty("question");
+            description.addProperty("actor");
+            description.addProperty("response");
+            description.addProperty("sourceId");
+            description.addProperty("sourceType");
+            description.addProperty("answeredTime");
+            description.addSelfLink();
+            return description;
+        } else if (rep instanceof RefRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("id");
+            description.addProperty("patient");
+            description.addProperty("person");
+            description.addProperty("actor");
+            description.addProperty("textQuestion");
+            description.addProperty("textResponse");
+            description.addProperty("question");
+            description.addProperty("response");
+            description.addSelfLink();
+            return description;
+        }
+        return null;
+    }
+    
+    /**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+    public DelegatingResourceDescription getCreatableProperties() {
```

Comment:
please look at the getCreateableProperties() and getUpdatableProperties() methods. They are almost identical and also very similar to above code. You are doing the same thing multiple times in whole file -> create a DelegatingResourceDescription object and add the same properties several times. Please extract repeated logic to separate method. There will be much less code and will be easier to read and understand.